Purchases/Public/RCPurchaserInfo.m

```diff
@@ -230,28 +224,41 @@ - (BOOL)isEqual:(RCPurchaserInfo *)other
     return isEqual;
 }
 
+- (NSDictionary *)descriptionDictionaryForEntitlementInfo:(RCEntitlementInfo *)info
+{
+    return @{
+             @"expiresDate": info.expirationDate ?: @"null",
+             @"latestPurchaseDate": info.latestPurchaseDate ?: @"null",
+             @"originalPurchaseDate": info.originalPurchaseDate ?: @"null",
+             @"periodType": info.periodType ? @(info.periodType) : @"null",
+             @"isActive": info.isActive ? @"Yes" : @"No",
+             @"willRenew": info.willRenew ? @"Yes" : @"No",
+             @"store": @(info.store),
+             @"productIdentifier": info.productIdentifier ?: @"null",
+             @"isSandbox": info.isSandbox ? @"Yes" : @"No",
+             @"unsubscribeDetectedAt": info.unsubscribeDetectedAt ?: @"null",
+             @"billingIssueDetectedAt": info.billingIssueDetectedAt ?: @"null"
+             };
+}
+
 - (NSString *)description
 {
     NSMutableDictionary *activeSubscriptions = [NSMutableDictionary dictionary];
     for (NSString *active in self.activeSubscriptions) {
         activeSubscriptions[active] = @{
-                                        @"expiresDate": [self expirationDateForProductIdentifier:active] ?: @"null"
+                                        @"expiresDate": [self expirationDateForProductIdentifier:active] ?: @"null",
                                         };
     }
 
     NSMutableDictionary *activeEntitlements = [NSMutableDictionary dictionary];
     for (NSString *active in self.entitlements.active) {
-        activeEntitlements[active] = @{
-                                        @"expiresDate": [self expirationDateForEntitlement:active] ?: @"null"
-                                        };
+        activeEntitlements[active] = [self descriptionDictionaryForEntitlementInfo:self.entitlements.active[active]];
```

Comment:
I would change `self.entitlements.active[active]` to `self.entitlements.active[activeEntitlement]` or something more clear